### PR TITLE
Add cloud-init timeout ansible module

### PR DESCRIPTION
**Pull request by julya935**

### CODE REVIEW

File: modules/cloud_init.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    - "Ansible module to manage cloud-init timeout"
+
+options:
+    cloud_init_timeout:
+      type: integer
+      display_name: "Cloud-init timeout"
+      ui_options:
+        advanced: true
+      required: false
+      description: "Timeout for cloud-init to finish running tasks, in seconds"
```
Comment (Maunty) [inline modules/cloud_init.py:36]: display_name?ui_options? what is this?

File: modules/cloud_init.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+"""
+
+
+EXAMPLES = """
+- name: "Wait for cloud-init to finish"
+  cloud_init:
+    cloud_init_timeout: 60
```
Comment (Maunty) [inline modules/cloud_init.py:43]: bad parameter name. Your module is allredy cloud_init

File: modules/cloud_init.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+"""
+
+import traceback
+import datetime
+from ansible.module_utils.basic import AnsibleModule
+
+
+def cloud_init_spec():
+    return dict(
+        cloud_init_timeout=dict(type="int", required=False, default=600)
```
Comment (Maunty) [inline modules/cloud_init.py:65]: same here as in docs.

File: modules/cloud_init.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+
+class CloudInit(AnsibleModule):
+    def __init__(self, **kwargs):
+        super(CloudInit, self).__init__(**kwargs)
+
+    def cloud_init_wait(self):
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=self.params.get("cloud_init_timeout"))
+        while datetime.datetime.now() < end_time:
+            _, stdout, stderr = self.run_command('cloud-init status')
```
Comment (Maunty) [inline modules/cloud_init.py:76]: add some sleep here. There is no need to ddos instance with this command. For example 5 sec sleep.

File: modules/cloud_init.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            if 'done' in stdout:
+                return stdout
+        raise Exception('Timeout exceeded')
```
Comment (Maunty) [inline modules/cloud_init.py:79]: Too base exception. Write some TimeoutException for this module.

File: modules/cloud_init.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        else:
+            response["msg"] = "Error during runtime occurred"
+            response["exception"] = traceback.format_exc()
+        module.fail_json(**response)
+
+    module.exit_json(**response)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (Maunty) [inline modules/cloud_init.py:103]: end of file. Do you use precommit?

**Review by Maunty — CHANGES_REQUESTED** [review]

File: modules/cloud_init.py (hunk 5 above)
Comment (julya935) [reply modules/cloud_init.py:79]: It seems like TimeoutError is not implemented in Python 2 https://github.com/hanxiao/bert-as-service/issues/427

Comment (Maunty) [reply modules/cloud_init.py:79]: Implement your own. Its not hard, for example:
https://github.com/arenadata/adcm_client/blob/9902732f7a23f16d8ef0968a40936b159d387b7e/adcm_client/packer/naming_rules.py#L21-L30

**Review by Maunty — APPROVED** [review]

**Pull request closed without merging**